src/paymentservice/src/main/java/ca/bc/gov/open/pssg/rsbc/dps/paymentservice/CalculateSinglePayment.java

```diff
@@ -19,65 +25,106 @@
-		  @RequestParam(value="ref2", required = false) String ref2,
-		  @RequestParam(value="ref3", required = false) String ref3,
-		  @RequestParam(value="minutesToExpire", required = true) String minutesToExpire) {
-	  
-	  //TODO - Complete main algorithm and logging 
-	  
-	  return new SinglePaymentResponse(PaymentServiceConstants.PAYMENT_SERVICE_RESP_MSG_OK, PaymentServiceConstants.PAYMENT_SERVICE_SUCCESS_CD, "https://mylongpaymenturl");
-	  
-  }
-  
-  /**
-   * handleMissingParams - Missing parameter handler. 
-   * 
-   * Returns legacy system equivalent response when required parameters missing from request. 
-   * 
-   * Note: This method, when invoked in the absence of a required parameter, will return an HTTP status code of 200. THIS
-   * IS EXPECTED AND WHAT THE LEGACY SYSTEM DOES. PLEASE DO NOT CHANGE THIS. 
-   * 
-   * @param ex
-   * @return
-   */
-  @ExceptionHandler(MissingServletRequestParameterException.class)
-  public SinglePaymentResponse handleMissingParams(MissingServletRequestParameterException ex) {
-      
-	  // TODO - Add appropriate logging here.  
-	  
-	  String paramName = ex.getParameterName();
-      // log.fatal (....
-     
-	 String errMsg = String.format(PaymentServiceConstants.PAYMENT_SERVICE_ERR_MISSING_PARAM, paramName);
-	  
-	 return new SinglePaymentResponse(errMsg, PaymentServiceConstants.PAYMENT_SERVICE_FAILURE_CD, null);
-      
-  }
+	@Value("${bambora.hostedpaymentendpoint}")
+	private String hostedPaymentEndpoint;
+
+	@Value("${bambora.merchantid}")
+	private String merchantId;
+
+	@Value("${bambora.hashkey}")
+	private String hashKey;
+
+	/**
+	 * 
+	 * singlepaymenturl
+	 * 
+	 * WARNING: Do not modify parameters. This operation must replicate legacy
+	 * system exactly.
+	 * 
+	 * Note: In the event of an exception, this methods will return an HTTP status
+	 * code of 200. THIS IS EXPECTED AND WHAT THE LEGACY SYSTEM DOES. PLEASE DO NOT
+	 * CHANGE THIS. An error is indicated to the calling system by the -1 status
+	 * code in the response structure.
+	 * 
+	 * @param transType
+	 * @param invoiceNumber
+	 * @param approvedPage
+	 * @param declinedPage
+	 * @param errorPage
+	 * @param totalItemsAmount
+	 * @param ref1
+	 * @param ref2
+	 * @param ref3
+	 * @param minutesToExpire
+	 * @return
+	 * @throws PaymentServiceException 
+	 */
+	@RequestMapping("/getSinglePaymentURL")
+	public SinglePaymentResponse singlepaymenturl(@RequestParam(value = "transType", required = true) String transType,
+			@RequestParam(value = "invoiceNumber", required = true) String invoiceNumber,
+			@RequestParam(value = "approvedPage", required = true) String approvedPage,
+			@RequestParam(value = "declinedPage", required = true) String declinedPage,
+			@RequestParam(value = "errorPage", required = true) String errorPage,
+			@RequestParam(value = "totalItemsAmount", required = true) String totalItemsAmount,
+			@RequestParam(value = "ref1", required = false) String ref1,
```

Comment:
do we need to provide required = false if the default is already false? someone of the examples I'm reading for @RequestParam don't include this.

Reply:
I just think it make things complete. I also have to add a bunch of other Swagger UI annotations so having these values makes it quicker for me.